### PR TITLE
fix(core/modal): fixes runtime error

### PR DESCRIPTION
**Pull request by matthiashader**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Unit tests (`yarn test`) were run locally and passed
- [x] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [x] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [x] Bug fix
- [ ] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

- In the preview example a runtime error occurs on react

GitHub Issue Number: #900

## What is the new behavior?

- no runtime error on the preview example

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by EnesKilicaslan** [comment]
Hi @matthiashader , thanks for the implementation.
However, I wonder how this will resolve the problem. The issue appears even I close the modal with cancel button.

**Comment by matthiashader** [comment]
> Hi @matthiashader , thanks for the implementation. However, I wonder how this will resolve the problem. The issue appears even I close the modal with cancel button.

Even if you remove the onCloseClick in the IxModalHeader? Like:
<Modal ref={modalRef}>
        <IxModalHeader>
          Message headline
        </IxModalHeader>

The error from my perspective was related to the fact that the modal-header and the also the event handling took care of closing the modal, so one of these didnt find it the before closed modal.

**Comment by EnesKilicaslan** [comment]
I see
At first, I thought it can not be related with header because I never close the modal from the header, instead I use cancel button.
However, it is not like that.
So, it seems it is fixed. Thanks

**Comment by danielleroux** [comment]
> I see At first, I thought it can not be related with header because I never close the modal from the header, instead I use cancel button. However, it might not be like that. All, I can say is it does not completely fix it.

What do you exactly mean?

File: packages/core/src/components/modal-header/modal-header.tsx [hunk 1]
```diff
@@ -55,7 +55,8 @@ export class ModalHeader {
   @Prop() iconColor: string;
 
   /**
-   * Close icon is clicked
+   * Fire Event when close icon is clicked and closes the modal
```
Comment (nuke-ellington) [inline packages/core/src/components/modal-header/modal-header.tsx:58]:
```suggestion
   * Emits when close icon is clicked and closes the modal
```

File: packages/core/src/components.d.ts [hunk 2]
```diff
@@ -4622,7 +4622,7 @@ declare namespace LocalJSX {
          */
         "iconColor"?: string;
         /**
-          * Close icon is clicked
+          * Fire Event when close icon is clicked and closes the modal Can be prevented, in which case only the event is triggered, and the modal remains open
```
Comment (nuke-ellington) [inline packages/core/src/components.d.ts:4625]:
```suggestion
          * Emits when close icon is clicked and closes the modal Can be prevented, in which case only the event is triggered, and the modal remains open
```

File: packages/core/component-doc.json [hunk 3]
```diff
@@ -8133,7 +8133,7 @@
           "bubbles": true,
           "cancelable": true,
           "composed": true,
-          "docs": "Close icon is clicked",
+          "docs": "Fire Event when close icon is clicked and closes the modal\nCan be prevented, in which case only the event is triggered, and the modal remains open",
```
Comment (nuke-ellington) [inline packages/core/component-doc.json:8136]:
```suggestion
          "docs": "Emits when close icon is clicked and closes the modal\nCan be prevented, in which case only the event is triggered, and the modal remains open",
```

File: packages/angular/src/components.ts [hunk 4]
```diff
@@ -1638,7 +1638,8 @@ export class IxModalHeader {
 
 export declare interface IxModalHeader extends Components.IxModalHeader {
   /**
-   * Close icon is clicked
+   * Fire Event when close icon is clicked and closes the modal
```
Comment (nuke-ellington) [inline packages/angular/src/components.ts:1641]:
```suggestion
   * Emits when close icon is clicked and closes the modal
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
JSDocs changes

**Comment by EnesKilicaslan** [comment]
> > I see At first, I thought it can not be related with header because I never close the modal from the header, instead I use cancel button. However, it might not be like that. All, I can say is it does not completely fix it.
> 
> What do you exactly mean?

I mean, just removing onCloseClick from ModalHeader does not solve the problem

**Comment by matthiashader** [comment]
> > > I see At first, I thought it can not be related with header because I never close the modal from the header, instead I use cancel button. However, it might not be like that. All, I can say is it does not completely fix it.
> > 
> > 
> > What do you exactly mean?
> 
> I mean, just removing onCloseClick from ModalHeader does not solve the problem

Could you then provide us with a snippet where the error still is thrown?

> > > I see At first, I thought it can not be related with header because I never close the modal from the header, instead I use cancel button. However, it might not be like that. All, I can say is it does not completely fix it.
> > 
> > 
> > What do you exactly mean?
> 
> I mean, just removing onCloseClick from ModalHeader does not solve the problem

Could you then provide us with a snippet where the error still remains? Because i can't produce any runtime error without the onCloseClick

File: packages/core/src/components/modal-content/modal-content.scss [hunk 5]
```diff
@@ -15,4 +15,5 @@
   position: relative;
   overflow: auto;
   padding: 0.125rem 0.625rem 0.125rem 0.625rem;
+  max-content: 60vh;
```
Comment (nuke-ellington) [inline packages/core/src/components/modal-content/modal-content.scss:18]: Not a valid CSS property? See e.g. [max-content](https://developer.mozilla.org/en-US/docs/Web/CSS/max-content).

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Minor change

File: packages/core/src/components/modal-content/modal-content.scss [hunk 6]
```diff
@@ -15,4 +15,5 @@
   position: relative;
   overflow: auto;
   padding: 0.125rem 0.625rem 0.125rem 0.625rem;
+  max-height: calc(80vh - 116px - 1em);
```
Comment (nuke-ellington) [inline packages/core/src/components/modal-content/modal-content.scss:18]: Full-size modals will not cover whole screen height with this max-height

**Review by nuke-ellington — DISMISSED** [review]

**Review by danielleroux — APPROVED** [review]

File: packages/core/src/components/modal-content/modal-content.scss (hunk 6 above)
Comment (jul-lam) [reply packages/core/src/components/modal-content/modal-content.scss]: Removed line because its not necessary.

**Review by danielleroux — DISMISSED** [review]

File: packages/core/src/components.d.ts [hunk 7]
```diff
@@ -5643,7 +5643,7 @@ declare namespace LocalJSX {
          */
         "iconColor"?: string;
         /**
-          * Close icon is clicked
+          * Emits when close icon is clicked and closes the modal Can be prevented, in which case only the event is triggered, and the modal remains open
```
Comment (jul-lam) [inline packages/core/src/components.d.ts:5646]: Missing new line

File: packages/angular/src/components.ts [hunk 8]
```diff
@@ -1719,7 +1719,8 @@ export class IxModalHeader {
 
 export declare interface IxModalHeader extends Components.IxModalHeader {
   /**
-   * Close icon is clicked
+   * Emits when close icon is clicked and closes the modal
+Can be prevented, in which case only the event is triggered, and the modal remains open
```
Comment (jul-lam) [inline packages/angular/src/components.ts:1723]: Missing "   * " before "Can ..."

File: packages/core/src/tests/modal/modal.e2e.ts [hunk 9]
```diff
@@ -85,16 +89,17 @@ test('modal with dropdown', async ({ mount, page }) => {
...
+          <ix-modal-header>Header</ix-modal-header>
+          <ix-modal-content>Some Content 123 content 123 content 123 content 123</ix-modal-content>
+          <ix-modal-footer>
+            <ix-button>Test</ix-button>
+          </ix-modal-footer>
+        </ix-modal>
+      `);
 
       const modal = page.locator('ix-modal');
+      await modal.evaluate((modal) => (modal as any).showModal());
```
Comment (jul-lam) [inline packages/core/src/tests/modal/modal.e2e.ts:102]: Why add another showModal()?

Comment (danielleroux) [reply packages/core/src/tests/modal/modal.e2e.ts:102]: Because thats the way how the modals work.

**Review by jul-lam — APPROVED** [review]

**Pull request merged**